packages/jsii/test/deprecation-warnings.test.ts

```diff
@@ -492,9 +642,207 @@ describe('Call injections', () => {
       { addDeprecationWarnings: true },
     );
 
-    expect(jsFile(result)).toMatch(
-      'constructor() { jsiiDeprecationWarnings.print("testpkg.Foo", "Use something else"); }',
+    expect(jsFile(result)).toMatchInlineSnapshot(`
+      "\\"use strict\\";
+      var _a;
+      Object.defineProperty(exports, \\"__esModule\\", { value: true });
+      exports.Foo = void 0;
+      const jsiiDeprecationWarnings = require(\\"./.warnings.jsii.js\\");
+      const JSII_RTTI_SYMBOL_1 = Symbol.for(\\"jsii.rtti\\");
+      /** @deprecated Use something else */
+      class Foo {
+          constructor() { try {
+              jsiiDeprecationWarnings.print(\\"testpkg.Foo\\", \\"Use something else\\");
+          }
+          catch (error) {
+              if (process.env.JSII_DEBUG !== \\"1\\" && error.name === \\"DeprecationError\\") {
+                  Error.captureStackTrace(error, this.constructor);
+              }
+              throw error;
+          } }
+      }
+      exports.Foo = Foo;
+      _a = JSII_RTTI_SYMBOL_1;
+      Foo[_a] = { fqn: \\"testpkg.Foo\\", version: \\"0.0.1\\" };
+      //# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJpbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7QUFDSSxxQ0FBcUM7QUFDckMsTUFBYSxHQUFHO0lBQ2Q7Ozs7Ozs7O09BQWU7O0FBRGpCLGtCQUVDIiwic291cmNlc0NvbnRlbnQiOlsiXG4gICAgLyoqIEBkZXByZWNhdGVkIFVzZSBzb21ldGhpbmcgZWxzZSAqL1xuICAgIGV4cG9ydCBjbGFzcyBGb28ge1xuICAgICAgY29uc3RydWN0b3IoKXt9XG4gICAgfVxuICAiXX0="
+    `);
+  });
+});
+
+describe('thrown exceptions have the expected stack trace', () => {
+  test('constructor', async () => {
+    const compilation = await compileJsiiForTest(
+      `
+      /** @deprecated for testing */
+      export class DeprecatedConstructor {
+        public constructor() {}
+      }
+
+      function test() {
+        new DeprecatedConstructor();
+      }
+
+      test();
+    `,
+      undefined,
+      { addDeprecationWarnings: true },
+    );
+    const source = jsFile(compilation);
+
+    const context = createVmContext(compilation);
+    try {
+      vm.runInContext(source, context, { filename: 'index.js' });
+      // The above line should have resulted in a DeprecationError being thrown
+      expect(null).toBeInstanceOf(Error);
```

Comment:
Not sure if I understand this line? Is the intention here to intentionally fail the test? Maybe something like this could make it clearer:
```suggestion
      throw new Error("expected a DeprecationError from running the transliterated snippet")
```
Or maybe "expect(true).toEqual(false)"

Reply:
The `expect(null)` check guarantees that the test will be marked as failed if no exception was thrown. If we move to the `throw` style, it'll trigger the catch and will happily update the snapshot when running with `-u`, which I think is not desirable behavior.

I kind of agree with your sentiment here, but this "hack" is just a bit safer.

Ideally - I'd have wanted to do `expect(() => vm.runInContext(...)).toThrow`, but that'll only verify the error message, not the stack trace (which is what I care about here).